sharedbuffers/mapped_struct.py

```diff
@@ -6326,7 +6595,14 @@ def build(cls, initializer, destfile = None, tempdir = None,
             del bigparts
 
             indexpos = curpos
-            write(buffer(index))
+            if python3:
+                if len(index):
+                    w = buffer(index)
+                else:
+                    w = b''
+            else:
+                w = buffer(index)
```

Comment:
This does not seem necessary. I'm perfectly able to do a memoryview of an empty array in py3. What exactly prompted this?

Reply:
It breaks when `index = array([], shape=(0, 2), dtype=uint64)`
```
File "sharedbuffers/mapped_struct.py", line 139, in sharedbuffers.mapped_struct.make_memoryview
    return memoryview(a).cast("B")
TypeError: memoryview: cannot cast view with zeros in shape or strides
```